examples/util/caffe2/Caffe2ModelImporter.js

```diff
@@ -0,0 +1,817 @@
+class Caffe2ModelImporter {
+  constructor (kwargs) {
+    this._isQuantized = kwargs.isQuantized;
+    this._rawModel = kwargs.rawModel;
+    this._inputSize = kwargs.inputSize;
+    this._isDNNL = kwargs.isDNNL;
+    this._model = null;
+    this._compilation = null;
+    this._execution = null;
+    this._tensorIds = [];         //{name: ID}
+    this._tensorTypes = [];       //{ID: type}
+    this._operations = [];        //{[opCode, inputs, outputs]}
+    this._operands = [];          //{ID: value}
+    this._quantParams = [];       //{ID: type}
+    this._requiredOps = new Set();
+    this._options = {
+      softmax: kwargs.softmax,
+    };
+    this._operandIndex = 0;
+    this._backend = kwargs.backend;
+    this._prefer = kwargs.prefer;
+    this._inputScaleFactor = kwargs.inputScaleFactor;
+
+    if (this._backend === 'WebML') {
+      if (nnNative === null) {
+        throw Error('Fails to initialize neural network context');
+      }
+      this._nn = nnNative;
+    } else if (this._backend === 'WASM' || this._backend === 'WebGL') {
+      this._nn = nnPolyfill;
+    }
+    this._bEagerMode = false;
+    this._supportedOps = new Set();
+  }
+
+  setEagerMode (flag) {
+    this._bEagerMode = flag;
+  };
+
+  setSupportedOps (ops) {
+    this._supportedOps = ops;
+  };
+
+  async createCompiledModel () {
+    let options = {
+      backend: this._backend,
+      eager: this._bEagerMode,
+      supportedOps: this._supportedOps,
+    };
+    this._model = await this._nn.createModel(options);
+    this._setInputTensor();
+    this._addOperandsAndArgs();
+    this._addInputsOutputs();
+
+    await this._model.finish();
+    this._compilation = await this._model.createCompilation();
+
+    let start = performance.now();
+    this._compilation.setPreference(getPreferCode(this._backend, this._prefer));
+    await this._compilation.finish();
+    this._execution = await this._compilation.createExecution();
+    let elapsed = performance.now() - start;
+    console.log(`compilation time: ${elapsed.toFixed(2)} ms`);
+  }
+
+  async compute (inputTensors, outputTensors) {
+    inputTensors.forEach((inputTensor, i) => {
+      this._execution.setInput(i, inputTensor);
+    });
+    outputTensors.forEach((outputTensor, i) => {
+      this._execution.setOutput(i, outputTensor);
+    });
+
+    let error = await this._execution.startCompute();
+    if (error) {
+      return error;
+    }
+    return 'success';
+  }
+
+  _setInputTensor () {
+    let inputName = this._rawModel[0].input[0].name;
+    let inputDims = this._inputSize.length == 3 ?
+    [1, this._inputSize[0], this._inputSize[1], this._inputSize[2]] : this._inputSize;
+    let inputType;
+    if (this._isQuantized) {
+      inputType = {
+        type: this._isDNNL ? this._nn.TENSOR_QUANT8_ASYMM_SIGNED : this._nn.TENSOR_QUANT8_ASYMM,
```

Comment:
We should use the value of zero_point to check the input type. 
`zero_point == 128?  this._nn.TENSOR_QUANT8_ASYMM_SIGNED : this._nn.TENSOR_QUANT8_ASYMM_SIGNED`